perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_login_namespace_umbrellaid_persistent_shadow.java

```diff
@@ -1,48 +1,137 @@
 package cz.metacentrum.perun.core.impl.modules.attributes;
 
-import cz.metacentrum.perun.core.implApi.modules.attributes.UserPersistentShadowAttributeWithConfig;
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.ExtSource;
+import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.UserExtSource;
+import cz.metacentrum.perun.core.api.exceptions.ExtSourceNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.UserExtSourceExistsException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.impl.modules.ModulesConfigLoader;
+import cz.metacentrum.perun.core.impl.modules.ModulesYamlConfigLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.Random;
+import java.util.UUID;
 
 /**
  * Class for checking logins uniqueness in the namespace and filling umbrellaid-persistent id.
  * It is only storage! Use module login umbrellaid_persistent for access the value.
  *
  */
 public class urn_perun_user_attribute_def_def_login_namespace_umbrellaid_persistent_shadow
-	extends UserPersistentShadowAttributeWithConfig {
+	extends urn_perun_user_attribute_def_def_login_namespace {
 
-	private final static String attrNameUmbrellaID = "login-namespace:umbrellaid-persistent-shadow";
+	private final static Logger log = LoggerFactory.getLogger(
+		urn_perun_user_attribute_def_def_login_namespace_umbrellaid_persistent_shadow.class);
 
 	private final static String CONFIG_EXT_SOURCE_NAME_UMBRELLA_ID = "extSourceNameUmbrellaID";
-	private final static String CONFIG_DOMAIN_NAME_UMBRELLA_ID = "domainNameUmbrellaID";
+	private final static String FRIENDLY_NAME = "login-namespace:umbrellaid-persistent-shadow";
+	private final static String FRIENDLY_NAME_PARAMETER = "umbrellaid-persistent-shadow";
+
+	private final ModulesConfigLoader loader = new ModulesYamlConfigLoader();
 
+	/**
+	 * fillAttribute will generate a UUID based on the current timestamp,
+	 * measured in units of 100 nanoseconds from the 15th of October 1582,
+	 * concatenated with the MAC address of the device where the UUID is created.
+	 */
 	@Override
-	public String getExtSourceConfigName() {
-		return CONFIG_EXT_SOURCE_NAME_UMBRELLA_ID;
+	public Attribute fillAttribute(PerunSessionImpl perunSession, User user, AttributeDefinition attribute) {
+
+		Attribute filledAttribute = new Attribute(attribute);
+
+		if (attribute.getFriendlyName().equals(FRIENDLY_NAME)) {
+			long most64SigBits = get64MostSignificantBits();
+			long least64SigBits = get64LeastSignificantBits();
+
+			filledAttribute.setValue(new UUID(most64SigBits, least64SigBits));
```

Comment:
I don't think you should set ``UUID`` object as attribute value. We support specific java types only and attribute is marked as ``String``, so maybe ``UUID.toString()`` ?

Reply:
Copy-paste mistake. Fixed.